src/odin/maneuversbuilder.cc

```diff
@@ -3939,5 +3943,25 @@ void ManeuversBuilder::CollapseMergeManeuvers(std::list<Maneuver>& maneuvers) {
   }
 }
 
+void ManeuversBuilder::AddLandmarksToManeuvers(std::list<Maneuver>& maneuvers) {
+  // the landmarks associated with edges in the previous maneuver. they should be added to the current
+  // maneuver
+  std::vector<Landmark> landmarks{};
+
+  for (auto man = maneuvers.begin(); man != maneuvers.end(); ++man) {
+    if (!landmarks.empty()) {
+      man->set_landmarks(landmarks);
+    }
+    // reset the landmarks to the ones associated with edges in the current maneuver
+    landmarks.clear();
+    // accumulate landmarks from all edges
+    for (auto node = man->begin_node_index(); node < man->end_node_index(); ++node) {
+      auto curr_edge = trip_path_->GetCurrEdge(node);
+      auto curr_landmarks = curr_edge->GetLandmarks();
+      std::move(curr_landmarks.begin(), curr_landmarks.end(), std::back_inserter(landmarks));
+    }
+  }
+}
+
```

Comment:
I'm not so sure if this is correct, but let's look at it together tmrw. My IDE breaks for odin for some reason, it's all undefined stuff, very hard to work with..

Reply:
I don't know if it's correct either. we don't have a test for this yet as I mentioned in our chat channel last Friday. let's see tomorrow.